lib/BackgroundJob/UnSnoozeJob.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2023 Johannes Merkel <mail@johannesgge.de>
+ *
+ * @author Johannes Merkel <mail@johannesgge.de>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Mail\BackgroundJob;
+
+use OCA\Mail\Account;
+use OCA\Mail\Contracts\IMailManager;
+use OCA\Mail\Db\MailAccountMapper;
+use OCA\Mail\Db\MailboxMapper;
+use OCA\Mail\Db\MessageMapper;
+use OCA\Mail\Db\MessageSnoozeMapper;
+use OCA\Mail\Exception\ClientException;
+use OCA\Mail\Exception\ServiceException;
+use OCA\Mail\IMAP\IMAPClientFactory;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\BackgroundJob\TimedJob;
+use Psr\Log\LoggerInterface;
+
+class UnSnoozeJob extends TimedJob {
+	public function __construct(
+		ITimeFactory $time,
+		private LoggerInterface $logger,
+		private IMAPClientFactory $clientFactory,
+		private MessageMapper $messageMapper,
+		private MessageSnoozeMapper $messageSnoozeMapper,
+		private MailAccountMapper $accountMapper,
+		private MailboxMapper $mailboxMapper,
+		private IMailManager $mailManager,
+	) {
+		parent::__construct($time);
+
+		$this->setInterval(60);
+		$this->setTimeSensitivity(self::TIME_SENSITIVE);
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function run($argument) {
+		$accounts = $this->accountMapper->getAllAccounts();
+		foreach ($accounts as $account) {
+			$account = new Account($account);
+
+			try {
+				$this->unsnoozeMails($account);
+			} catch (ServiceException|ClientException $e) {
+				$this->logger->error('Could not unSnooze Mails', [
+					'exception' => $e,
+					'userId' => $account->getUserId(),
+					'accountId' => $account->getId(),
+					'snoozeMailboxId' => $account->getMailAccount()->getSnoozeMailboxId(),
+				]);
+			}
+		}
+
+	}
+
+	/**
+	 * @throws ClientException
+	 * @throws ServiceException
+	 */
+	private function unSnoozeMails(Account $account): void {
+		$snoozeMailboxId = $account->getMailAccount()->getSnoozeMailboxId();
+		if ($snoozeMailboxId === null) {
+			return;
+		}
+
+		try {
+			$snoozeMailbox = $this->mailboxMapper->findById($snoozeMailboxId);
+		} catch (DoesNotExistException $e) {
+			return;
+		}
+
+		$now = $this->time->getTime();
+		$messages = $this->messageMapper->findMessagesToUnSnooze(
+			$snoozeMailboxId,
+			$now,
```

Comment:
the job runs once an hour. if you snooze a message until *tomorrow morning, 9am* and the job always runs at :45, the message will be woken at 9:45 only.

idea: consider 30min look-ahead for the background job. then messages are woken +-30min of the target date

Reply:
the job runs every 60 *seconds*. there is no delay.